Feature/video stream

agatheblues opened the pull request:
Fixed : 

- Camel cases / constants 
- Move UDP related method to UDP Client class
- Remove unused methods or references
- Thread management

Note: beach ball animation is commented out temporarily

Commit:
Add animation class for video stream that extends the base animation

Commit:
Include UDP Video stream Client class, and create video stream class which displays the images from the stream on the canvas

Commit:
Add settings to boost brightness and saturation of the image

Commit:
Adjust settings

Commit:
Modify comment of VideoStream

Commit:
Remove IPClient constants as it is not used

Commit:
Move socket related code to the UDP class

Commit:
Fix visibility and camel cases variables

Commit:
Fix constants and camel cases (2)

Commit:
Move method to convert buffered image to Pimage in the udp video stream client class

Commit:
Move all images methods to the UDP stream client class, change width and aspect ratio of expected image

Commit:
Check if packets come from the expected IP

Commit:
Remove method to boost colors as it is possible to do it on the camera itself

Commit:
Remove unneeded references to brightness and saturation settings

Commit:
Comment out the beach ball animation temporarily

Commit:
Fix thread issue une video stream: currentThread() corresponds to the Processing sketch thread, not the udp client thread

wilcotomassen commented on client/settings.json:
```diff
@@ -2,5 +2,6 @@
 	"videoDir": "",
-	"imageDir": ""
+	"imageDir": "",
+	"IPSERVER": ""
```
Can we change this to ipServer, or something so that the casing of all settings is consistent? Not that there's anything inherently wrong with this, but just inconsistent with the existing settings

agatheblues replied:
Agreed & fixed

wilcotomassen commented:
Some issues to address:

- I think currently it's not really multithreaded in the sense that the capturing of packets is done independently of the VideoStream class. I think the UDPVideoStreamClient is waiting until the getStreamImage method is called, only then does it look at potential incoming packets. Ideally it would continuously accept the packets, store it in the streamImage (overwriting the previous) and then serving the latest image when getStreamImage is called. To do this, I think you should use the run method (see here: https://bitbucket.org/cleverfranke/livingcell/src/6ba77412fc215656865b04f92ef9a5f2e1d06bc9/RGL/src/system/blob/NetworkBlobReader.java?at=develop&fileviewer=file-view-default)
- The UDPVideoStreamClient could be idle when the VideoStream class is not the active Animation. To make that happen the VideoStream should implement the isStarting and isStopping methods, and enable/disable the UDP client accordingly
- A lot of the members in UDPVideoStreamClient are public even though the feel as unlogical to call from the outside consider making them private
- The overall sequence of fetching an UDP packet and creating a PImage from it (which is in the end what you'd want out of it), doesn't become clear immediately.  A lot of in between variables are class members. Could there be one method that does the entire sequence, with all those variables encapsulated, potentially using static methods to modify them so that the class becomes more modular?

Less important, but still relevant
- Indentation is all over the place, possibly also because of mixed spaces and tabs. We should standardize and then make indentation correct everywhere. Now it hurts my head. It's ok to keep this now, then we can do a big "fixed all whitespace" change.

Commit:
Change casing of IPSERVER setting

Commit:
Change visibility of UDPVideoStreamClient to private when it is relevant

Commit:
Do not exit application in case of SocketException, just print stack trace

Commit:
Refactor class members and initalizations of variables. Move the check for empty IPSERVER provided

Commit:
Make the udp client continuously accept packet, and the animation only getting the image. Refactor the receivePacket() method.

Commit:
Stop Udp client if it's not the current animation, start it when animation is selected

Commit:
Clean settings.json

agatheblues commented:
- Not multithreaded: implemented the run() method, which receives packet independantly of the VideoStream animation
- Idle when not active animation: implemented the isStarting and isStopping method. I would have ideally initialize the socket and thread in the constructor and then connect/disconnect and start/stop in these method but it seems complex to do this.
- Visibility: All class members and variables are now private except start(), stop(), run() and getImage()
- Sequence of UDP packet: created sub method that are more encapsulated
- Quit the application: Removed it! 
- IPSERVER : changed it to ipServer
- Indentation: Same here, it's difficult to see what are the _real_ changes. Ok to do the whitespace big change later!

Commit:
Fix access to ipServer value: add default value
